tests/unit/pipeline/test_ner_re_pipeline.py

```diff
@@ -0,0 +1,390 @@
+
+    # from predictions
+
+    prepared_docs = add_annotations_from_other_documents(
+        docs=[document],
+        other_docs=[document_with_relations],
+        layer_names=["binary_relations"],
+        from_predictions=True,
+    )
+
+    prepared_doc = prepared_docs[0]
+    relations = prepared_doc["binary_relations"]
+    assert len(relations) == 1
+
+    # to predictions
+
+    prepared_docs = add_annotations_from_other_documents(
+        docs=[document],
+        other_docs=[document_with_relations],
+        layer_names=["binary_relations"],
+        to_predictions=True,
+    )
+
+    prepared_doc = prepared_docs[0]
+    relations = prepared_doc["binary_relations"].predictions
+    assert len(relations) == 3  # adds to previously existing predictions
+
+    # inplace
+
+    prepared_docs = add_annotations_from_other_documents(
+        docs=[document],
+        other_docs=[document_with_relations],
+        layer_names=["binary_relations"],
+        inplace=True,
+    )
+
+    relations = document["binary_relations"]
+    assert len(relations) == 2
+
+
+@dataclasses.dataclass
+class TextDocumentWithEntitiesAndRelations(TextBasedDocument):
+    entities: AnnotationLayer[LabeledSpan] = annotation_field(target="text")
+    relations: AnnotationLayer[BinaryRelation] = annotation_field(target="entities")
+
+
+@pytest.fixture
+def old_document():
+    document = TextDocumentWithEntitiesAndRelations(
+        text="Harry lives in Berlin, Germany. He works at DFKI.", id="tmp"
+    )
+
+    document.entities.extend(
+        [
+            LabeledSpan(start=0, end=5, label="PERSON"),
+            LabeledSpan(start=15, end=30, label="LOCATION"),
+            LabeledSpan(start=44, end=48, label="ORGANIZATION"),
+        ]
+    )
+    assert str(document.entities[0]) == "Harry"
+    assert str(document.entities[1]) == "Berlin, Germany"
+    assert str(document.entities[2]) == "DFKI"
+
+    document.relations.extend(
+        [
+            BinaryRelation(
+                head=document.entities[0],
+                tail=document.entities[1],
+                label="lives_in",
+            ),
+            BinaryRelation(
+                head=document.entities[0],
+                tail=document.entities[2],
+                label="works_at",
+            ),
+        ]
+    )
+
+    return document
+
+
+@pytest.mark.slow
+def test_ner_re_pipeline(old_document):
+    pipeline = NerRePipeline(
+        ner_model_path="pie/example-ner-spanclf-conll03",
+        re_model_path="pie/example-re-textclf-tacred",
+        entity_layer="entities",
+        relation_layer="relations",
+        device=-1,
+        batch_size=1,
+        show_progress_bar=False,
+    )
+
+    docs = pipeline(documents=[old_document])
+    assert len(docs) == 1
+
+    doc = docs[0]
+    predicted_entities = doc.entities.predictions
+    assert len(predicted_entities) == 4
+    predicted_relations = doc.relations.predictions
+    assert len(predicted_relations) == 0
```

Comment:
test the predicted annotations, i.e. entity texts, labels, and relation argument texts